R/dependencies.R

```diff
@@ -0,0 +1,410 @@
+##' @title Get the dependencies for a given report from the database
+##'
+##' @param name the name of the report
+##' @param id the id of the report, if omitted, use the id of the latest report
+##' @param upstream A boolean indicating if we want to move up or down the tree
+##' @param con A connection to a database
+get_dependencies_db <- function(name, id, upstream, con, list_all = FALSE) {
+  ## now construct the SQL query
+  if (upstream) {
+    filt_qry <- sprintf("depends.report_version='%s'", id)
+  } else {
+    filt_qry <- sprintf("report_version.id='%s'", id)
+  }
+
+  sql_qry <- paste(c("SELECT",
```

Comment:
Can we move this into the sql directory in inst? If not, there seem to be more commas here than needed